qcelemental/models/align.py

```diff
@@ -126,6 +126,27 @@ def align_hessian(self, hess) -> NDArray:
         alhess = blockwise_contract(alhess)
         return alhess
 
+    def align_vector_gradient(self, mu_x, mu_y, mu_z):
+        """Align the nuclear gradients of vector components (e.g. dipole derivatives)."""
+        # Input data is assumed to be organized into separate x, y, z vector components.
+        # Organize derivatives for each atom into 3x3 and transform it.
+        nat = mu_x.shape[0]
+        al_mu_x = np.zeros( (nat, 3) )
+        al_mu_y = np.zeros( (nat, 3) )
+        al_mu_z = np.zeros( (nat, 3) )
+
+        Datom = np.zeros( (3,3) ) # atom whose nuclear derivatives are taken
+        for at in range(nat):
+            Datom.fill(0)
+            Datom[0,:] = mu_x[at,:]
+            Datom[1,:] = mu_y[at,:]
+            Datom[2,:] = mu_z[at,:]
+            Datom[:] = np.dot( self.rotation.T , np.dot(Datom, self.rotation) )
+            al_mu_x[self.atommap[at]][:] = Datom[0,:]
+            al_mu_y[self.atommap[at]][:] = Datom[1,:]
+            al_mu_z[self.atommap[at]][:] = Datom[2,:]
+        return (al_mu_x, al_mu_y, al_mu_z)
```

Comment:
since inputs and outputs are xyz components, would it make sense to either do three separate calls or pass as [`(3 * nat, 3)`](https://github.com/psi4/psi4/blob/41f824cee253fc4262c08d7d0ae22190a889eae0/psi4/src/psi4/libmints/wavefunction.h#L245)? like to take better advantage of broadcasting.

Reply:
What I like about the 3 explicit arguments is that it is obvious to any user what should be passed, particularly for a mixed-derivative like this - versus the challenge of a new person (or myself in 6 months:) trying to figure out what the single variable data structure should be.  For the Hessian you would guess correctly, maybe not for this.  But I'm fine changing it if you think there's a compelling reason and relying on examples, tests, and documentation.

Reply:
sorry for the delay -- it's been a crazy month. talking it over, we think the packed arrangement would be more in keeping with the other functions. the first line of the fn can be `mu_x, mu_y, mu_z = *mu` and the return can be `return al_mu` and the test call can be `p2c_dipder_x, p2c_dipder_y, p2c_dipder_z = mill.align_vector_gradient(...)`